homeassistant/components/event/__init__.py

```diff
@@ -154,7 +155,17 @@ def _trigger_event(
     ) -> None:
         """Process a new event."""
         if event_type not in self.event_types:
-            raise ValueError(f"Invalid event type {event_type} for {self.entity_id}")
+            event_types: str = ", ".join(self.event_types)
+            raise HomeAssistantError(
```

Comment:
I'm not sure about this change. This is not a user facing error. It's an error caused by the implementor of the event platform. That the MQTT integration allows users to implement the platform incorrectly is a problem that the MQTT integration should solve.

Reply:
> This is not a user facing error.

That is true...

Opening a revert

Reply:
Ah inderd.

Reply:
But if it happens, wouldn't it be cleaner to catch it? Though it’s not a user facing error, the user is inconvenienced by an issue of the platform. With the multitude of platforms wouldnt it be better to catch the error so a user can properly report on it?

Reply:
It is still caught and logged. Just not translated.